diff --git a/code/clusterer/src/python/paraprobe_clusterer.py b/code/clusterer/src/python/paraprobe_clusterer.py
index 3ea51d1..d14ac1a 100644
--- a/code/clusterer/src/python/paraprobe_clusterer.py
+++ b/code/clusterer/src/python/paraprobe_clusterer.py
@@ -25,13 +25,13 @@ import h5py
 from pytictoc import TicToc
 from joblib import cpu_count
 from ifes_apt_tc_data_modeling.utils.utils import hash_to_isotope
-from utils.src.python.iontype_handling import IonCloud, IonType
-from utils.src.python.numerics \
+from paraprobe_utils.iontype_handling import IonCloud, IonType
+from paraprobe_utils.numerics \
     import MAX_NUMBER_OF_ATOMS_PER_MOLECULAR_ION, RMX, MYHDF5_COMPRESSION_DEFAULT
-from clusterer.src.python.paraprobe_configclusterer import ConfigClusterer
-from utils.src.python.hdf_five_string_io import write_strings_to_attribute
-from utils.src.python.hashing import get_simulation_id, get_file_hashvalue
-from utils.src.python.gitinfo import PARAPROBE_VERSION, NEXUS_VERSION
+from paraprobe_clusterer.paraprobe_configclusterer import ConfigClusterer
+from paraprobe_utils.hdf_five_string_io import write_strings_to_attribute
+from paraprobe_utils.hashing import get_simulation_id, get_file_hashvalue
+from paraprobe_utils.gitinfo import PARAPROBE_VERSION, NEXUS_VERSION
 
 
 use_gpu = False
diff --git a/code/clusterer/src/python/paraprobe_configclusterer.py b/code/clusterer/src/python/paraprobe_configclusterer.py
index 2f4e250..5bf7d4a 100644
--- a/code/clusterer/src/python/paraprobe_configclusterer.py
+++ b/code/clusterer/src/python/paraprobe_configclusterer.py
@@ -19,7 +19,7 @@ import numpy as np
 import h5py
 # from pytictoc import TicToc
 # from ifes_apt_tc_data_modeling.utils.nist_isotope_data import isotopes
-from utils.src.python.hdf_five_string_io import read_strings_from_dataset
+from paraprobe_utils.hdf_five_string_io import read_strings_from_dataset
 
 
 class AlgorithmDBScan():
diff --git a/code/parmsetup/src/python/clusterer_config.py b/code/parmsetup/src/python/clusterer_config.py
index a20e239..67bab4b 100644
--- a/code/parmsetup/src/python/clusterer_config.py
+++ b/code/parmsetup/src/python/clusterer_config.py
@@ -21,9 +21,9 @@ from string import digits
 import numpy as np
 
 from ifes_apt_tc_data_modeling.utils.utils import symbol_lst_to_matrix_of_nuclide_vector
-from utils.src.python.numerics import EPSILON
-from utils.src.python.nodeinfo import NodeInfo
-from utils.src.python.supertool import ParmsetupBase, ParmsetupTaskBase
+from paraprobe_utils.numerics import EPSILON
+from paraprobe_utils.nodeinfo import NodeInfo
+from paraprobe_utils.supertool import ParmsetupBase, ParmsetupTaskBase
 
 
 
diff --git a/code/parmsetup/src/python/distancer_config.py b/code/parmsetup/src/python/distancer_config.py
index fbe27ae..4f35a00 100644
--- a/code/parmsetup/src/python/distancer_config.py
+++ b/code/parmsetup/src/python/distancer_config.py
@@ -18,10 +18,10 @@
 
 import numpy as np
 
-from utils.src.python.numerics import EPSILON
-from utils.src.python.nodeinfo import NodeInfo
-from utils.src.python.supertool import ParmsetupBase, ParmsetupTaskBase
-from utils.src.python.hashing import get_file_hashvalue
+from paraprobe_utils.numerics import EPSILON
+from paraprobe_utils.nodeinfo import NodeInfo
+from paraprobe_utils.supertool import ParmsetupBase, ParmsetupTaskBase
+from paraprobe_utils.hashing import get_file_hashvalue
 
 
 class PointToTriangleSetDistancing(ParmsetupTaskBase):
diff --git a/code/parmsetup/src/python/intersector_config.py b/code/parmsetup/src/python/intersector_config.py
index a4e1361..7b82d0f 100644
--- a/code/parmsetup/src/python/intersector_config.py
+++ b/code/parmsetup/src/python/intersector_config.py
@@ -21,10 +21,10 @@ from string import digits
 
 import numpy as np
 
-from utils.src.python.numerics import EPSILON
-from utils.src.python.hashing import get_file_hashvalue
-from utils.src.python.nodeinfo import NodeInfo
-from utils.src.python.supertool import ParmsetupBase, ParmsetupTaskBase
+from paraprobe_utils.numerics import EPSILON
+from paraprobe_utils.hashing import get_file_hashvalue
+from paraprobe_utils.nodeinfo import NodeInfo
+from paraprobe_utils.supertool import ParmsetupBase, ParmsetupTaskBase
 
 
 class VolumeFeatureSubSet:
diff --git a/code/parmsetup/src/python/nanochem_config.py b/code/parmsetup/src/python/nanochem_config.py
index edd6ab5..df1b46e 100644
--- a/code/parmsetup/src/python/nanochem_config.py
+++ b/code/parmsetup/src/python/nanochem_config.py
@@ -22,13 +22,13 @@ import numpy as np
 from os.path import exists
 from ase.data import atomic_numbers, chemical_symbols
 
-from utils.src.python.numerics import \
+from paraprobe_utils.numerics import \
     TOOLS_IN_TOOLBOX, EPSILON, MAX_NUMBER_OF_ATOMS_PER_MOLECULAR_ION
-from utils.src.python.hashing import get_file_hashvalue
-from utils.src.python.nodeinfo import NodeInfo
-from utils.src.python.supertool import ParmsetupBase, ParmsetupTaskBase
-from utils.src.python.arg_validation import validate_argument_str
-from utils.src.python.primscontinuum import RoiRotatedCylinder
+from paraprobe_utils.hashing import get_file_hashvalue
+from paraprobe_utils.nodeinfo import NodeInfo
+from paraprobe_utils.supertool import ParmsetupBase, ParmsetupTaskBase
+from paraprobe_utils.arg_validation import validate_argument_str
+from paraprobe_utils.primscontinuum import RoiRotatedCylinder
 from ifes_apt_tc_data_modeling.utils.utils import \
     get_smart_chemical_symbols, isotope_to_hash
 from ifes_apt_tc_data_modeling.utils.nist_isotope_data import isotopes
diff --git a/code/parmsetup/src/python/ranger_config.py b/code/parmsetup/src/python/ranger_config.py
index db8533c..87b430f 100644
--- a/code/parmsetup/src/python/ranger_config.py
+++ b/code/parmsetup/src/python/ranger_config.py
@@ -16,7 +16,7 @@
 #
 """Create NeXus/HDF5 configuration files for paraprobe-ranger."""
 
-from utils.src.python.supertool import ParmsetupBase, ParmsetupTaskBase
+from paraprobe_utils.supertool import ParmsetupBase, ParmsetupTaskBase
 
 
 class ApplyExistentRanging(ParmsetupTaskBase):
diff --git a/code/parmsetup/src/python/selector_config.py b/code/parmsetup/src/python/selector_config.py
index 255d86c..bc55e5a 100644
--- a/code/parmsetup/src/python/selector_config.py
+++ b/code/parmsetup/src/python/selector_config.py
@@ -16,7 +16,7 @@
 #
 """Create NeXus/HDF5 configuration files for paraprobe-selector."""
 
-from utils.src.python.supertool import ParmsetupBase, ParmsetupTaskBase
+from paraprobe_utils.supertool import ParmsetupBase, ParmsetupTaskBase
 
 
 class RoiSelectionTask(ParmsetupTaskBase):
diff --git a/code/parmsetup/src/python/spatstat_config.py b/code/parmsetup/src/python/spatstat_config.py
index 6ee647d..2d7a1aa 100644
--- a/code/parmsetup/src/python/spatstat_config.py
+++ b/code/parmsetup/src/python/spatstat_config.py
@@ -21,10 +21,10 @@ from string import digits
 import numpy as np
 
 from ifes_apt_tc_data_modeling.utils.utils import symbol_lst_to_matrix_of_nuclide_vector
-from utils.src.python.numerics import EPSILON
-from utils.src.python.hashing import get_file_hashvalue
-from utils.src.python.nodeinfo import NodeInfo
-from utils.src.python.supertool import ParmsetupBase, ParmsetupTaskBase
+from paraprobe_utils.numerics import EPSILON
+from paraprobe_utils.hashing import get_file_hashvalue
+from paraprobe_utils.nodeinfo import NodeInfo
+from paraprobe_utils.supertool import ParmsetupBase, ParmsetupTaskBase
 
 
 class SpatstatTask(ParmsetupTaskBase):
diff --git a/code/parmsetup/src/python/surfacer_config.py b/code/parmsetup/src/python/surfacer_config.py
index bd35848..9e94122 100644
--- a/code/parmsetup/src/python/surfacer_config.py
+++ b/code/parmsetup/src/python/surfacer_config.py
@@ -18,9 +18,9 @@
 
 import numpy as np
 
-from utils.src.python.numerics import EPSILON
-from utils.src.python.nodeinfo import NodeInfo
-from utils.src.python.supertool import ParmsetupBase, ParmsetupTaskBase
+from paraprobe_utils.numerics import EPSILON
+from paraprobe_utils.nodeinfo import NodeInfo
+from paraprobe_utils.supertool import ParmsetupBase, ParmsetupTaskBase
 
 
 class SurfaceMeshingTask(ParmsetupTaskBase):
diff --git a/code/parmsetup/src/python/tessellator_config.py b/code/parmsetup/src/python/tessellator_config.py
index 26ced7a..de4ec07 100644
--- a/code/parmsetup/src/python/tessellator_config.py
+++ b/code/parmsetup/src/python/tessellator_config.py
@@ -18,9 +18,9 @@
 
 import numpy as np
 
-from utils.src.python.hashing import get_file_hashvalue
-from utils.src.python.nodeinfo import NodeInfo
-from utils.src.python.supertool import ParmsetupBase, ParmsetupTaskBase
+from paraprobe_utils.hashing import get_file_hashvalue
+from paraprobe_utils.nodeinfo import NodeInfo
+from paraprobe_utils.supertool import ParmsetupBase, ParmsetupTaskBase
 
 
 class TessellationTask(ParmsetupTaskBase):
diff --git a/code/parmsetup/src/python/transcoder_config.py b/code/parmsetup/src/python/transcoder_config.py
index 582a9bd..efb8845 100644
--- a/code/parmsetup/src/python/transcoder_config.py
+++ b/code/parmsetup/src/python/transcoder_config.py
@@ -16,9 +16,9 @@
 #
 """Create NeXus/HDF5 configuration files for paraprobe-transcoder."""
 
-from utils.src.python.hashing import get_file_hashvalue
-from utils.src.python.nodeinfo import NodeInfo
-from utils.src.python.supertool import ParmsetupBase, ParmsetupTaskBase
+from paraprobe_utils.hashing import get_file_hashvalue
+from paraprobe_utils.nodeinfo import NodeInfo
+from paraprobe_utils.supertool import ParmsetupBase, ParmsetupTaskBase
 
 
 class TranscodingTask(ParmsetupTaskBase):
diff --git a/code/reporter/src/python/nanochem_report.py b/code/reporter/src/python/nanochem_report.py
index 4c2118d..af236cc 100644
--- a/code/reporter/src/python/nanochem_report.py
+++ b/code/reporter/src/python/nanochem_report.py
@@ -22,12 +22,12 @@ import numpy as np
 import h5py
 from ase.data import chemical_symbols, atomic_numbers
 
-from utils.src.python.hdf_five_string_io import read_strings_from_dataset
-from utils.src.python.bisection import bisection
+from paraprobe_utils.hdf_five_string_io import read_strings_from_dataset
+from paraprobe_utils.bisection import bisection
 # from utils.src.python.filteredmean import filtered_mean
-import reporter.src.python.plotting.default.scatter_plot_property_set_over_value_set as mysct
-import reporter.src.python.plotting.default.line_plot_property_set_for_fixed_value as mycdf
-import reporter.src.python.plotting.default.composition_plots as mycmp
+import paraprobe_reporter.plotting.default.scatter_plot_property_set_over_value_set as mysct
+import paraprobe_reporter.plotting.default.line_plot_property_set_for_fixed_value as mycdf
+import paraprobe_reporter.plotting.default.composition_plots as mycmp
 from ifes_apt_tc_data_modeling.utils.utils import isotope_to_hash
 
 
diff --git a/code/reporter/src/python/spatstat_report.py b/code/reporter/src/python/spatstat_report.py
index 814d611..f637094 100644
--- a/code/reporter/src/python/spatstat_report.py
+++ b/code/reporter/src/python/spatstat_report.py
@@ -19,8 +19,8 @@
 import numpy as np
 import h5py
 
-from utils.src.python.numerics import EPSILON
-import reporter.src.python.plotting.default.line_plot_property_set_for_fixed_value as mycdf
+from paraprobe_utils.numerics import EPSILON
+import paraprobe_reporter.plotting.default.line_plot_property_set_for_fixed_value as mycdf
 
 
 class ReporterSpatstat():
diff --git a/code/reporter/src/python/tessellator_report.py b/code/reporter/src/python/tessellator_report.py
index ed33aa5..e56b1f4 100644
--- a/code/reporter/src/python/tessellator_report.py
+++ b/code/reporter/src/python/tessellator_report.py
@@ -19,7 +19,7 @@
 import numpy as np
 import h5py
 
-import reporter.src.python.plotting.default.line_plot_property_set_for_fixed_value as mycdf
+import paraprobe_reporter.plotting.default.line_plot_property_set_for_fixed_value as mycdf
 
 
 class ReporterTessellator():
diff --git a/code/tessellator/CMakeLists.txt b/code/tessellator/CMakeLists.txt
index 6d401d2..0a8cce7 100644
--- a/code/tessellator/CMakeLists.txt
+++ b/code/tessellator/CMakeLists.txt
@@ -126,16 +126,16 @@ add_executable(${MYPROJECTNAME}
 	${COMPILEPATH}/PARAPROBE_TriangleSoupHdl.cpp.o
 	${COMPILEPATH}/PARAPROBE_ToolsBaseHdl.cpp.o
 
-	${MYVOROSRCPATH}/c_loops.cc.o
-	${MYVOROSRCPATH}/cell.cc.o
-	${MYVOROSRCPATH}/common.cc.o
-	${MYVOROSRCPATH}/container.cc.o
-	${MYVOROSRCPATH}/container_prd.cc.o
-	${MYVOROSRCPATH}/pre_container.cc.o
-	${MYVOROSRCPATH}/unitcell.cc.o
-	${MYVOROSRCPATH}/v_base.cc.o
-	${MYVOROSRCPATH}/v_compute.cc.o
-	${MYVOROSRCPATH}/wall.cc.o
+	${MYVOROSRCPATH}/c_loops.cc
+	${MYVOROSRCPATH}/cell.cc
+	${MYVOROSRCPATH}/common.cc
+	${MYVOROSRCPATH}/container.cc
+	${MYVOROSRCPATH}/container_prd.cc
+	${MYVOROSRCPATH}/pre_container.cc
+	${MYVOROSRCPATH}/unitcell.cc
+	${MYVOROSRCPATH}/v_base.cc
+	${MYVOROSRCPATH}/v_compute.cc
+	${MYVOROSRCPATH}/wall.cc
 
 	${MYTOOLSRCPATH}/PARAPROBE_ConfigTessellator.cpp
 	${MYTOOLSRCPATH}/PARAPROBE_TessellatorStructs.cpp
diff --git a/code/utils/src/python/primscontinuum.py b/code/utils/src/python/primscontinuum.py
index 6a10152..1fdc670 100644
--- a/code/utils/src/python/primscontinuum.py
+++ b/code/utils/src/python/primscontinuum.py
@@ -18,7 +18,7 @@
 
 import numpy as np
 
-from utils.src.python.numerics import EPSILON
+from paraprobe_utils.numerics import EPSILON
 
 
 class RoiRotatedCuboid:
diff --git a/code/utils/src/python/supertool.py b/code/utils/src/python/supertool.py
index 5b45f94..1b88ad1 100644
--- a/code/utils/src/python/supertool.py
+++ b/code/utils/src/python/supertool.py
@@ -23,16 +23,16 @@ import datetime as dt
 import numpy as np
 import h5py
 
-from utils.src.python.numerics import SUPPORTED_DATATYPES, TOOLS_IN_TOOLBOX
-from utils.src.python.primscontinuum \
+from paraprobe_utils.numerics import SUPPORTED_DATATYPES, TOOLS_IN_TOOLBOX
+from paraprobe_utils.primscontinuum \
     import RoiRotatedCuboid, RoiRotatedCylinder, RoiSphere
-from utils.src.python.nodeinfo import NodeInfo
-from utils.src.python.nexus import specific_nxdl_path, \
+from paraprobe_utils.nodeinfo import NodeInfo
+from paraprobe_utils.nexus import specific_nxdl_path, \
     nxdl_path_resolve_parent_of_attribute, \
     nxdl_path_resolve_name_of_attribute
-from utils.src.python.hashing import get_file_hashvalue
-from utils.src.python.gitinfo import PARAPROBE_VERSION, NEXUS_VERSION
-from utils.src.python.hdf_five_string_io \
+from paraprobe_utils.hashing import get_file_hashvalue
+from paraprobe_utils.gitinfo import PARAPROBE_VERSION, NEXUS_VERSION
+from paraprobe_utils.hdf_five_string_io \
     import write_strings_to_attribute, write_strings_to_dataset
 
 
